sycl/source/detail/device_info.hpp

```diff
@@ -568,64 +568,108 @@ struct get_device_info_impl<range<Dimensions>,
   }
 };
 
+// This macro is only for AMD, NVIDIA and Intel GPU architectures
+#define ARCHES(X)                                                              \
```

Comment:
It turns out that the way the `uint32_t` keys were converted was wrong anyways. The converted strings were not in hexadecimal form anymore and that's also why the CI failed.
I've added another macro for now to avoid the conversion and leave the DeviceIP as an integer.